Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -1317,4 +1317,118 @@ void delete_records_from_an_object_store(GC::Ref<ObjectStore> store, GC::Ref<IDB
     // 3. Return undefined.
 }
 
+// https://w3c.github.io/IndexedDB/#store-a-record-into-an-object-store
+WebIDL::ExceptionOr<GC::Ptr<Key>> store_a_record_into_an_object_store(JS::Realm& realm, GC::Ref<ObjectStore> store, JS::Value value, GC::Ptr<Key> key, bool no_overwrite)
+{
+    // 1. If store uses a key generator, then:
+    if (store->key_generator().has_value()) {
+        // 1. If key is undefined, then:
+        if (key == nullptr) {
+            // 1. Let key be the result of generating a key for store.
+            auto maybe_key = generate_a_key(realm, store);
```

Comment:
eh. nit, but you could probably put 1. and 2. over an `auto key = TRY(generate_a_key(realm, store));` and not step on any toes. well except the ones related to the fact that generate_a_key returns ... an integer? rather than a `Key`